src/source/src/parser/avro_parser/schema_resolver.rs

```diff
@@ -112,24 +112,31 @@ impl ConfluentSchemaResolver {
         let schema = Schema::parse_str(&cf_schema.raw)
             .map_err(|e| RwError::from(ProtocolError(format!("Avro schema parse error {}", e))))?;
         let resolver = ConfluentSchemaResolver {
-            writer_schemas: SkipMap::new(),
+            writer_schemas: Cache::new(u64::MAX),
             confluent_client: client,
         };
-        resolver.writer_schemas.insert(cf_schema.id, schema.clone());
+        resolver
+            .writer_schemas
+            .insert(cf_schema.id, Arc::new(schema.clone()))
+            .await;
         Ok((schema, resolver))
     }
 
     // get the writer schema by id
-    pub async fn get(&self, schema_id: i32) -> Result<Entry<'_, i32, Schema>> {
-        if let Some(entry) = self.writer_schemas.get(&schema_id) {
-            Ok(entry)
+    pub async fn get(&self, schema_id: i32) -> Result<Arc<Schema>> {
+        if let Some(schema) = self.writer_schemas.get(&schema_id) {
```

Comment:
We can always use `get_with` here. :)

Reply:
`confluent_client.get_schema_by_id` and `Schema::parse_str` could generate err and we can't throw it out in a clousure.